src/adagrams.js

```diff
@@ -1,15 +1,87 @@
+import { 
+  LETTER_POOL,
+  LETTER_SCORE,
+} from "constants.js";
+
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let letterPoolCopy = {...LETTER_POOL};
+  let hand = [];
+  while (hand.length < 10) {
+    let letter = Object.keys(letterPoolCopy)[Math.floor(Math.random()*Object.keys(letterPoolCopy).length)];
+    hand.push(letter);
+    letterPoolCopy[letter]--;
+    if (letterPoolCopy[letter] === 0) {
+      delete letterPoolCopy[letter];
+    }
+  }
+  return hand;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const lettersInHandCopy = [...lettersInHand];
+  const inputArray = Array.from(input);
+
+  if (input.length < 1 || input.length > 10) return false;
```

Comment:
Consider having line 28 be at the start of the function. That allows us to not waste any time with the other operations if we're passed invalid input. Also consider throwing an exception here instead of returning false.